src/Event.cpp

```diff
@@ -25,52 +25,85 @@ void Event::add_value(std::string name, std::string&& arg) {
     string_values_.emplace(name, arg);
 }
 
+//get_value
+//Params: name of event, address to store result at, bool for error checking. if true and an error occurs then exit program
 template<>
-int Event::get_value(const std::string& name, int otherwise) const {
+bool Event::get_value(const std::string& name, std::string* value, bool crash_on_fail) const {
+    const auto val = string_values_.find(name); //val - pointer to the event "name"
+
+    if (val == string_values_.end()) {
+        std::cout << "Value " << name << " not found in event " << static_cast<int>(event_type) << "!" << std::endl;
+
+        if (crash_on_fail == true) {   //if there is an error and crash is set to true crash program
+            std::cerr << "Error: Value not found and program will crash on fail.";
+            exit(EXIT_FAILURE);
```

Comment:
As discussed in group chat, use asserts to crash program instead of exiting for all get_value implementations

Reply:
not sure how to use asserts to crash, can you give me an example?